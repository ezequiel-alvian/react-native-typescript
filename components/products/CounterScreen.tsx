import React from 'react'
import { View, Text, StyleSheet, Dimensions } from 'react-native'
import { Button } from '@rneui/themed'
import { IProps } from '../../types/types'

const CounterScreen: React.FC<IProps> = ({count, setCount})=> {

    const handleIcrease = () => setCount(count + 1)
    const handleDecrease = () => {
        if (count === 0) return 
        setCount(count - 1) 
    }

    return(
        <View style={styles.containerButton}>
            <Button 
                onPress={handleIcrease}
                titleStyle={{color:'#000'}} 
                buttonStyle={styles.buttonCustom}
            >+</Button>
            <View>
                <Text>{count}</Text>
            </View>
            <Button
                onPress={handleDecrease}
                titleStyle={{color:'#000'}} 
                buttonStyle={styles.buttonCustom}
            >-</Button>
        </View>
    )
}

const styles = StyleSheet.create({
    containerButton: {
        display:'flex',
        flexDirection:'row',
        justifyContent:'space-between',
        alignItems:'center',
        backgroundColor:'#00FF7F',
        width: (Dimensions.get('window').width -20 ),
    },
    buttonCustom: {
        backgroundColor:'#00FF7F',
    },
})

export default CounterScreen