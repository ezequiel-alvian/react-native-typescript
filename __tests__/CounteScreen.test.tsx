import CounterScreen from "../components/products/CounterScreen"
import { render } from '@testing-library/react-native'

describe('<CounterScreen/>', ()=>{
    const data = {
        count:0,
        setCount:jest.fn()
    }
    let component
    test('CounterScreen', ()=>{
        component=render(<CounterScreen count={data.count} setCount={data.setCount} />)
        expect(component.toJSON()).toMatchSnapshot()
    })
})