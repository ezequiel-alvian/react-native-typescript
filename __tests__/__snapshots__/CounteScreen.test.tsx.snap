// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterScreen/> CounterScreen 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#00FF7F",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": 730,
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 2,
        },
        undefined,
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": false,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#00FF7F",
            "borderColor": "#2089dc",
            "borderRadius": 2,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
  <View>
    <Text>
      0
    </Text>
  </View>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 2,
        },
        undefined,
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": false,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#00FF7F",
            "borderColor": "#2089dc",
            "borderRadius": 2,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          -
        </Text>
      </View>
    </View>
  </View>
</View>
`;
